cmake/Modules/FindTPLONEDPL.cmake

```diff
@@ -0,0 +1,14 @@
+INCLUDE(CheckIncludeFileCXX)
+CHECK_INCLUDE_FILE_CXX(oneapi/dpl/execution KOKKOS_COMPILER_HAS_ONEDPL_EXECUTION_HEADER)
+CHECK_INCLUDE_FILE_CXX(oneapi/dpl/algorithm KOKKOS_COMPILER_HAS_ONEDPL_ALGORITHM_HEADER)
+
+IF (KOKKOS_COMPILER_HAS_ONEDPL_EXECUTION_HEADER AND KOKKOS_COMPILER_HAS_ONEDPL_ALGORITHM_HEADER)
+  KOKKOS_CREATE_IMPORTED_TPL(ONEDPL INTERFACE)
+ELSE()
+  FIND_PACKAGE(oneDPL REQUIRED)
+  KOKKOS_CREATE_IMPORTED_TPL(
+    ONEDPL INTERFACE
+    LINK_LIBRARIES oneDPL
+    COMPILE_DEFINITIONS PSTL_USE_PARALLEL_POLICIES=0 _GLIBCXX_USE_TBB_PAR_BACKEND=0
```

Comment:
I am not super familiar with this, but is there an alternative?

Reply:
If you hit the problem because you are using a gcc-9 or gcc-10 toolchain, there isn't much else users can do (apart from reordering include files which is brittle). The question is whether we should try to fix it by disabling the TBB backend for functions in `<algorithm>` or let the user figure it out themselves in case they are affected.

Reply:
I'm OK with this since codes that are using `Kokkos` are probably not interested in using parallel stl but we need at the very least a `FIMXE` so we can remove that at some point.

Reply:
Please confirm that these definition are always needed regardless of the toolchain that is being used.
If the answer is no, I would prefer if we only added them when absolutely necessary,

Reply:
As I said in https://github.com/kokkos/kokkos/pull/5229#discussion_r953731053, you need this if the `<algorithm>` header comes from `gcc-9` or `gcc-10`. Since we can't know at configure-time what toolchain users will use in their code, I would just do it unconditionally. (If we are assuming they are using the same toolchain for everything, we could base the decision on `_GLIBCXX_RELEASE` but that seems more fragile to me.)

Reply:
It's more fragile than assuming the same compiler was used and I'm trying to say that adding a configure-time check(https://github.com/kokkos/kokkos/pull/5229/commits/08b90bb3ef0e872c971dab7f57fb0e7e18f2c60f) adds more fragility to it. Anyway, I added it and also made sure that we add the definitions if the compiler already includes `oneDPL` implicitly.

Reply:
It was a legitimate question, not me asking to change.
BTW would you remind me if we consider using a push_macro pragma.

Reply:
I tried and couldn't make it work.